File: 1580-shuffle-the-array/shuffle-the-array.cpp
class Solution {
public:
    vector<int> shuffle(vector<int>& nums, int n) {
        vector<int>ans;
        ans.reserve(2*n);
        int first =0;
        int second =n;
        int idx=0;
        while(first<n||second<2*n){
            ans.push_back(nums[first]);
            ans.push_back(nums[second]);
            first++, second++;
        }
        return ans;
    }
};

File: 1580-shuffle-the-array/shuffle-the-array.java
class Solution {
    public int[] shuffle(int[] nums, int n) {
        
        int i = 0;
        int j = n;
        int idx = 0;
        int[]  ans  = new int[2*n];
        while(i<n && j<(2*n) ){
            ans[idx] = nums[i];
            i++;
            idx++;
            ans[idx]=nums[j];
            idx++;
            j++;
        }
        return ans;
    }
}